pkg/user/external.go

```diff
@@ -0,0 +1,273 @@
+package user
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/dotmesh-io/dotmesh/pkg/types"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type ExternalManager struct {
+	url string
+}
+
+func NewExternal(url string) *ExternalManager {
+	return &ExternalManager{
+		url: url,
+	}
+}
+
+func (m *ExternalManager) call(operation string, method string, body interface{}, result interface{}) error {
+	client := &http.Client{}
```

Comment:
this is a memory leak as each http.Client creates a goroutine for tcp keepalives. This way you also miss the transport that has inbuilt HTTP_PROXY env variable support. 

Do something like:

```
type ExternalManager struct {
	url string
        httpClient *http.Client
}

func NewExternal(url string, httpClient *http.Client) *ExternalManager {
        if httpClient == nil {
           httpClient = http.DefaultClient
        }
	return &ExternalManager{
		url: url,
                httpClient: httpClient,
	}
}
```

Reply:
Good call, ta!